libs/ngxtension/inject-query-params/src/inject-query-params.ts

```diff
@@ -3,55 +3,269 @@ import { toSignal } from '@angular/core/rxjs-interop';
+ */
+export function injectQueryParams(
+	key?: string,
+	options?: QueryParamsOptions<string, string, string>,
+): Signal<string | null>;
+
+/**
+ * The `injectQueryParams` function allows you to access and manipulate query parameters from the current route.
+ * It retrieves the value of a query parameter based on a custom transform function applied to the query parameters object.
+ *
+ * @template ReadT - The expected type of the read value.
+ * @param {QueryParamsTransformFn<ReadT>} fn - A transform function that takes the query parameters object (`params: Params`) and returns the desired value.
+ * @returns {Signal} A `Signal` that emits the transformed value based on the provided custom transform function.
+ *
+ * @example
+ * const searchValue = injectQueryParams((params) => params['search'] as string);
  */
-export function injectQueryParams<T>(
-	transform: (params: Params) => T,
-): Signal<T>;
+export function injectQueryParams<ReadT>(
+	fn: QueryParamsTransformFn<ReadT>,
+): Signal<ReadT>;
 
 /**
- * Injects the query params from the current route.
- * If a key is provided, it will return the value of that key.
- * If a transform function is provided, it will return the result of that function.
- * Otherwise, it will return the entire query params object.
+ * The `injectQueryParams` function allows you to access and manipulate query parameters from the current route.
+ *
+ * @template ReadT - The expected type of the read value.
+ * @param {string} keyOrParamsTransform - The name of the query parameter to retrieve, or a transform function to apply to the query parameters object.
+ * @param {QueryParamsOptions} options - Optional configuration options for the query parameter.
+ * @returns {QueryParamsOptions} A `Signal` that emits the transformed value of the specified query parameter, or the entire query parameters object if no key is provided.
  *
  * @example
  * const search = injectQueryParams('search'); // returns the value of the 'search' query param
  * const search = injectQueryParams(p => p['search'] as string); // same as above but can be used with a custom transform function
+ * const idParam = injectQueryParams('id', {transform: numberAttribute}); // returns the value fo the 'id' query params and transforms it into a number
+ * const idParam = injectQueryParams(p => numberAttribute(p['id'])); // same as above but can be used with a custom transform function
  * const queryParams = injectQueryParams(); // returns the entire query params object
- *
- * @param keyOrTransform OPTIONAL The key of the query param to return, or a transform function to apply to the query params object
  */
-export function injectQueryParams<T>(
-	keyOrTransform?: string | ((params: Params) => T),
-): Signal<T | Params | string | null> {
+export function injectQueryParams<ReadT>(
+	keyOrParamsTransform?: string | ((params: Params) => ReadT),
+	options?: QueryParamsOptions<ReadT, string, ReadT>,
```

Comment:
```suggestion
	options: QueryParamsOptions<ReadT, string, ReadT> = {},
```
It would remove the condition a few lines below.